documentation/smg/16_serialiser.md

```diff
@@ -0,0 +1,229 @@
+_2_    | 64 + type  | dim1 | dim2 |     | Data
+_N_    | N<<5 + type| dim1 | dim2 | ... | Data
+
+Where `data` is the concatenation of the serialisation of each element.
+
+#### Object Array
+Object array data (type tag 26) are serialised as:
+
+_Rank_ |     tag    | dim1 | dim2 | ... |Data|     ...            |    ...     |   ...   |  ...
+-------|------------|------|------|:---:|----|--------------------|------------|---------|------
+_NULL_ | 32 + type  | 0    |      |     | 33 | length(class_name) | class_name |         |
+_0_    |    type    |      |      |     | 33 | length(class_name) | class_name | ser_tag | Data |
+_1_    | 32 + type  | nElem|      |     | 33 | length(class_name) | class_name | ser_tag | Data |
+_2_    | 64 + type  | dim1 | dim2 |     | 33 | length(class_name) | class_name | ser_tag | Data |
+_N_    | N<<5 + type| dim1 | dim2 | ... | 33 | length(class_name) | class_name | ser_tag | Data |
+
+Where the first 3 elements of data 
+(`33`,`length(class_name)`,`class_name`) are a serialised representation of the class name (see [Simple data][simpledata]),
+`ser_tag` is a `uint8` tag (described below) and
+`data` is the serialisation of the object by means described by `ser_tag`
+
+###### ser_tag
+The object array's `ser_tag` contains information about the means by which the object has been serialised.
+It can have one of three values:
+
+0) The object has been serialised by its own `serialize` method
+1) The object has been serialised by the `saveobj` method or function
+2) The object has been serialised by calling `struct` on the object
+
+#### Function handles
+Function handles are serialised differently depending on their type.
+As function handles in MATLAB are purely scalar objects, the rank-component of the tag is utilised for tagging the type of function handle (thus the method to [de]serialise), these types are:
+
+Handle type         |   tag
+--------------------|--------
+Simple/Class simple | 32 + 25
+Anonymous           | 64 + 25
+Scoped              | 96 + 25
+
+Function handles are serialised as:
+Handle type         |   tag   | data                                       | ...
+--------------------|---------|--------------------------------------------|-----------------------------------------
+Simple/Class simple |  57     | function name as serialised string         |
+Anonymous           |  89     | anonymous function as serialised string    | relevant workspace as serialised struct
+Scoped              |  121    | function parentage as serialised cell array|
+
+### Limitations
+- Java objects cannot be serialized.
+- Cannot handle arrays of more than rank-7.
+- Cannot handle arrays where the size of any dimension > (2^32)-1 due to serialising dimensions as `uint32`.
+  __NB.__ This includes sparse matrices.
+- Handles to nested/scoped functions can only be deserialized when their parent functions
+  support the BCILAB argument reporting protocol (e.g., by using arg_define).
```

Comment:
What is the BCILAB argument reporting protocol? Can you add a reference to some documentation

Reply:
Turns out it __is__ the brain-computer interface stuff, seems they've done some general MATLAB improvements in their branch.

https://sccn.ucsd.edu/wiki/Arg_define